planseqlearn/psl/mp_env.py

```diff
@@ -97,115 +98,88 @@ def get_object_pose_mp(self, obj_idx=0):
                 get_intermediate_frames=get_intermediate_frames,
             )
         obs = self.get_observation()
-        self.teleport_on_grasp = True
-        self.teleport_on_place = False
-        self.num_high_level_steps += 1
         return obs
 
     def step(self, action, get_intermediate_frames=False, **kwargs):
         o, r, d, i = self._wrapped_env.step(action)
-        is_grasped = self.check_grasp()
-        open_gripper_on_tp = False
-        if self.teleport_on_grasp:
-            take_planner_step = is_grasped
-            if take_planner_step:
-                self.teleport_on_grasp = False
-                self.teleport_on_place = True
-        elif self.teleport_on_place:
-            # object we are checking is the number of high level steps we have taken so far
-            placed = self.check_object_placement(obj_idx=self.object_idx)
-            take_planner_step = (
-                placed
-                and not is_grasped
-                and not self.check_robot_collision(ignore_object_collision=False)
-            )
-            if take_planner_step:
-                self.object_idx += 1
-                self.object_idx = min(self.object_idx, len(self.text_plan) // 2 - 1)
-                if self.num_high_level_steps < len(self.text_plan) - 1:
-                    self.curr_obj_name = self.text_plan[self.num_high_level_steps][0]
-                open_gripper_on_tp = True
-                self.teleport_on_place = False
-                self.teleport_on_grasp = True
-        if take_planner_step:
-            if self.env_name.startswith("NutAssembly") or self.env_name.startswith(
-                "PickPlace"
-            ):
-                for i in range(len(self.placement_poses)):
-                    if i % 2 == 0:
-                        pos, obj_quat = self.get_object_pose_mp(obj_idx=i)
-                        pos += np.array([0.0, 0.0, self.vertical_displacement])
-                        self.compute_hardcoded_orientation(pos, obj_quat)
-            if (self.num_high_level_steps // 2) >= len(self.placement_poses):
-                take_planner_step = False
-        if take_planner_step:
+        take_planner_step = self.curr_postcondition() and (self.curr_plan_stage) != len(self.text_plan) - 1
+        if take_planner_step: # advance plan to next stage using motion planner
+            self.curr_plan_stage += 1
             target_pos, target_quat = self.get_target_pos()
             if self.teleport_instead_of_mp:
                 error = self.set_robot_based_on_ee_pos(
                     target_pos.copy(),
                     target_quat.copy(),
                     self.reset_qpos,
                     self.reset_qvel,
-                    is_grasped=is_grasped,
-                    obj_idx=self.object_idx,
-                    open_gripper_on_tp=open_gripper_on_tp,
+                    obj_name=self.text_plan[self.curr_plan_stage - (self.curr_plan_stage % 2)][0],
```

Comment:
why this complicated formula for indexing the text plan? if the text plan is (obj, post cond), ... then the index should just be the current plan stage right?

Reply:
this i guess is a bit of an artifact of the implementation - basically since all functions are now indexed by object name, in set_robot_..., all the get_object_pose, if using the current stage, would cause changes in the pose of (eg) the gold peg, when the correct thing would be to change the gold nut, and this basically indexes back by 1 in that case

Reply:
maybe leave this for now, but consider implementing things in a way that you can use the current stage as the index, this indexing is a bit hard to understand